src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Collection/IAsyncEnumerableOfTConverter.cs

```diff
@@ -81,10 +91,11 @@ protected override bool OnWriteResume(Utf8JsonWriter writer, TAsyncEnumerable va
                 }
             }
 
+            Debug.Assert(moveNextTask.IsCompleted);
```

Comment:
In a resume case, can `IsCompleted` be `false`?

Reply:
No. If the converter has been suspended, then root async context would await any pending tasks.